src/core/src/buffer.rs

```diff
@@ -103,6 +103,8 @@ pub enum Role {
     Index,
     /// Constant buffer
     Constant,
+    /// Staging buffer
+    Staging,
```

Comment:
What happens when a buffer is not used for anything but staging ? It does not make much sense for it to be any of the current roles.